api/controllers/StatsController.js

```diff
@@ -121,18 +121,30 @@ function addMatchToRankings(season, match, player, opponent) {
  * }
  */
 function transformSeasonToDTO(season) {
-  const { rankings, ...rest } = season;
+  let { rankings, gameCounts, uniquePlayersPerWeek, ...rest } = season;
+
   // Convert rankings from Map to Array
-  const rankingsAsArray = Array.from(rankings.values()).map((player) => {
+  rankings = Array.from(rankings.values()).map((player) => {
     return {
       ...player,
       matches: Object.fromEntries(player.matches), // Convert matches from Map to Object
     };
   });
+
+  // Remove zero-valued gameCounts + uniquePlayersPerWeek from the end
+  if (gameCounts[gameCounts.length - 1] === 0) {
+    gameCounts.pop();
+    uniquePlayersPerWeek.pop();
+  }
+
+  // Convert uniquePlayersPerWeek from Set[] to int[] number of players
+  uniquePlayersPerWeek = uniquePlayersPerWeek.map((playerSet) => playerSet.size);
+
   return {
     ...rest,
-    rankings: rankingsAsArray,
-    uniquePlayersPerWeek: season.uniquePlayersPerWeek.map((playerSet) => playerSet.size),
+    rankings,
+    gameCounts,
+    uniquePlayersPerWeek,
   };
```

Comment:
One note is I think we could alternatively skip `...rest` and simply `return season` since we change data in place now. That might be cleanest

Reply:
Update: I tried this and it pisses off the linter because the mutated variables are "unused". That's not really true in practice but I'm not looking to fight the linter on such a small issue